iotdb/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -18,116 +18,604 @@
+import org.apache.iotdb.tsfile.file.metadata.enums.CompressionType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.iotdb.tsfile.read.common.Field;
 import org.apache.iotdb.tsfile.read.common.Path;
-import org.apache.iotdb.tsfile.read.expression.QueryExpression;
+import org.apache.iotdb.tsfile.read.common.RowRecord;
+import org.apache.iotdb.tsfile.read.expression.IExpression;
 import org.apache.iotdb.tsfile.read.query.dataset.QueryDataSet;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.utils.Pair;
 
-public abstract class QueryProcessExecutor implements IQueryProcessExecutor {
+public class QueryProcessExecutor extends AbstractQueryProcessExecutor {
 
-  protected ThreadLocal<Integer> fetchSize = new ThreadLocal<>();
-  protected IEngineQueryRouter queryRouter = new EngineQueryRouter();
+  private StorageEngine storageEngine;
+  private MManager mManager = MManager.getInstance();
 
-  @Override
-  public QueryDataSet processQuery(PhysicalPlan queryPlan, QueryContext context)
-      throws IOException, FileNodeManagerException, PathErrorException,
-      QueryFilterOptimizationException, ProcessorException {
-
-    if (queryPlan instanceof QueryPlan) {
-      return processDataQuery((QueryPlan) queryPlan, context);
-    } else if (queryPlan instanceof AuthorPlan) {
-      return processAuthorQuery((AuthorPlan) queryPlan, context);
-    } else {
-      throw new ProcessorException(String.format("Unrecognized query plan %s", queryPlan));
-    }
+  public QueryProcessExecutor() {
+    storageEngine = StorageEngine.getInstance();
   }
 
-  protected abstract QueryDataSet processAuthorQuery(AuthorPlan plan, QueryContext context)
-      throws ProcessorException;
-
-  private QueryDataSet processDataQuery(QueryPlan queryPlan, QueryContext context)
-      throws FileNodeManagerException, QueryFilterOptimizationException, PathErrorException, ProcessorException, IOException {
-    QueryExpression queryExpression = QueryExpression.create().setSelectSeries(queryPlan.getPaths())
-        .setExpression(queryPlan.getExpression());
-    if (queryPlan instanceof GroupByPlan) {
-      GroupByPlan groupByPlan = (GroupByPlan) queryPlan;
-      return groupBy(groupByPlan.getPaths(), groupByPlan.getAggregations(),
-          groupByPlan.getExpression(), groupByPlan.getUnit(), groupByPlan.getOrigin(),
-          groupByPlan.getIntervals(), context);
+  public static String checkValue(TSDataType dataType, String value) throws ProcessorException {
+    if (dataType == TSDataType.BOOLEAN) {
+      value = value.toLowerCase();
+      if (SQLConstant.BOOLEAN_FALSE_NUM.equals(value)) {
+        value = "false";
+      } else if (SQLConstant.BOOLEAN_TRUE_NUM.equals(value)) {
+        value = "true";
+      } else if (!SQLConstant.BOOLEN_TRUE.equals(value) && !SQLConstant.BOOLEN_FALSE
```

Comment:
you do not need to check "else if".
By the way, because SQLConstant.BOOLEAN_FALSE_NUM and SQLConstant.BOOLEAN_TRUE_NUM are 0 and 1, you do not need to execute `value = value.toLowerCase`

Reply:
fixed